rajawali/src/main/java/c/org/rajawali3d/core/AFrameDelegate.java

```diff
@@ -0,0 +1,234 @@
+                if (callback.callFrameStart()) {
+                    frameStartCallbacks.add(callback);
+                }
+                if (callback.callFrameEnd()) {
+                    frameEndCallbacks.add(callback);
+                }
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean removeFrameCallback(@NonNull final FrameCallback callback) {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                if (callback.callFrameStart()) {
+                    frameStartCallbacks.remove(callback);
+                }
+                if (callback.callFrameEnd()) {
+                    frameEndCallbacks.remove(callback);
+                }
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean clearFrameCallbacks() {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                frameStartCallbacks.clear();
+                frameEndCallbacks.clear();
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean addAnimation(final Animation anim) {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                animations.add(anim);
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean removeAnimation(final Animation anim) {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                animations.remove(anim);
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean replaceAnimation(final Animation oldAnim, final Animation newAnim) {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                animations.set(animations.indexOf(oldAnim), newAnim);
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean addAnimations(final Collection<Animation> anims) {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                animations.addAll(anims);
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean clearAnimations() {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                animations.clear();
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    //
+    // ACoreComponent method overrides
+    //
+
+    @RenderThread
+    @Override
+    @CallSuper
+    public void onFrameStart(double deltaTime) throws InterruptedException {
+        if (!isEnabled) {
```

Comment:
The issue comes from, as a developer, you expect stacktraces of classes that extend to call super first so without expecting every single implementation life cycle events become difficult to understand when they don't happen in the natural order of `let the parent do the work before I do my work`

I believe in most situations like this @jwoolston and I would typically agree on is try to use a delegate instead of extending a method to hack in life cycle changes. I'm not sure how or if that can apply here.